### ISSUE TITLE
Authentication Error while pulling images via docker-maven-plugin

### ISSUE THREAD

**Issue by vishalvsh1** [spotify/docker-maven-plugin#246]
## Description

Bug - Plugin expects that image which Dockerfile is pulling needs to be passed with username and password, which should not an essential parameter as most of the images does not need any auth to be downloaded. 
## How to reproduce
1. Remove all the images from local.
2. Run the docker based build (mvn clean insall).

This should run the build without any auth error.
## What happened instead

[INFO] Building image your-application:0.0.3
Step 1 : FROM ubunt:latest
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] your-application..................................... FAILUER [  3.619 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 39.938 s (Wall Clock)
[INFO] Finished at: 2016-06-25T16:26:06+00:00
[INFO] Final Memory: 65M/1216M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.spotify:docker-maven-plugin:0.3.7:build (default) on project your-application: Exception caught: Get https://registry-1.docker.io/v2/ubuntu/manifests/latest: unauthorized: incorrect username or password -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
## Software:
- docker version: 1.10
- docker-maven-plugin version: 0.3.7
- maven version: 3.4
## Full backtrace

``` text
[Paste full backtrace here]
```

**Comment by vishalvsh1** [comment]
I have written a blog too for same error.

Here you can find more detail.
http://www.scmtechblog.net/2016/06/authentication-error-while-pulling.html

**Comment by mattnworb** [comment]
> Plugin expects that image which Dockerfile is pulling needs to be passed with username and password

No, that shouldn't really be the case, and I don't think that the plugin is expecting that. 

Can you post your pom.xml (or at least the configuration for the plugin)? I'd like to see if there is any [authentication](https://github.com/spotify/docker-maven-plugin#authenticating-with-private-registries) setup.

Have you tried using the most recent version of the plugin?

**Comment by vishalvsh1** [comment]
Following is the sample configuration from my pom file.
```
<plugin>
				<groupId>com.spotify</groupId>
				<artifactId>docker-maven-plugin</artifactId>
				<version>0.4.1</version>
				<configuration>
					<imageName>ourprivateregistry.com:5002/application:${project.version}</imageName>

					<serverId>docker</serverId>
					<dockerDirectory>${basedir}</dockerDirectory>
					<resources>
						<resource>
							<targetPath>/</targetPath>
							<directory>${project.build.directory}</directory>
							<include>${project.build.finalName}.war</include>
						</resource>
					</resources>
				</configuration>
				<executions>
					<execution>
						<id>build-image</id>
						<phase>package</phase>
						<goals>
							<goal>build</goal>
						</goals>
					</execution>
				</executions>

			</plugin>
```

**Comment by timwebster9** [comment]
I'm seeing this as well.  I am also using a private registry (Nexus 3), and I have <serverId> set , configured with its credentials in settings.xml.  If you remove <serverId> it works, but then you can't push to your private registry. @vishalvsh1's workaround works, but it's not ideal.

**Comment by adakappagari** [comment]
Please fix this issue. We are running into the same one when job attempts to pull the dependent images.

**Comment by papasmurf17** [comment]
I've the same issue, please fix it!

**Comment by s17t** [comment]
This is still an issue in 0.4.13, any change to see this fixed ?

**Comment by Jinshui** [comment]
Will this issue get fixed? I am using nexus 3 as my private docker registry.  And the following configuration is not working while directly running 'docker pull my-private-registry/image-name' works fine.

<useConfigFile>true</useConfigFile>

**Comment by Jinshui** [comment]
I am using nexus3 as the private docker registry, and using the 'useConfigFile' property for authentication, the following configuration just works fine for me:

`
    <properties>
        <docker.image>network-service</docker.image>
        <docker.registry>xxxxxxxxx</docker.registry>
        <docker.registry.port>9082</docker.registry.port>
    </properties>
            
            <plugin>
                <groupId>com.spotify</groupId>
                <artifactId>docker-maven-plugin</artifactId>
                <version>0.4.13</version>
                <configuration>
                    <dockerDirectory>${project.build.directory}/deploy/docker</dockerDirectory>
                    <registryUrl>${docker.registry}</registryUrl>
                    <imageName>${docker.registry}:${docker.registry.port}/${docker.image}</imageName>
                    <imageTags>
                        <imageTag>${project.version}</imageTag>
                    </imageTags>
                    <resources>
                        <resource>
                            <targetPath>/</targetPath>
                            <directory>${project.build.directory}</directory>
                            <include>${project.build.finalName}.jar</include>
                        </resource>
                        <resource>
                            <targetPath>/</targetPath>
                            <directory>${project.build.directory}/deploy/docker</directory>
                            <include>application.yaml</include>
                        </resource>
                    </resources>
                    <useConfigFile>true</useConfigFile>
                </configuration>
                <executions>
                    <execution>
                        <id>build-image</id>
                        <goals>
                            <goal>build</goal>
                        </goals>
                        <phase>package</phase>
                    </execution>
                    <execution>
                        <id>tag-image</id>
                        <goals>
                            <goal>tag</goal>
                        </goals>
                        <phase>package</phase>
                        <configuration>
                            <image>${docker.registry}:${docker.registry.port}/${docker.image}:${project.version}</image>
                            <newName>${docker.registry}:${docker.registry.port}/${docker.image}:${project.version}</newName>
                        </configuration>
                    </execution>
                    <execution>
                        <id>push-image</id>
                        <goals>
                            <goal>push</goal>
                        </goals>
                        <phase>deploy</phase>
                        <configuration>
                            <imageName>${docker.registry}:${docker.registry.port}/${docker.image}:${project.version}</imageName>
                        </configuration>
                    </execution>
                </executions>
            </plugin>`

**Comment by aniruthmp** [comment]
Referring to [Private Docker Repo #1439](https://github.com/docker/for-mac/issues/1439),  I updated my keychain to https://some_web.com and I was able to successfully run docker build -f Dockerfile
But spotify/docker-maven-plugin fails
`[ERROR] Failed to execute goal com.spotify:docker-maven-plugin:0.4.13:build (default) on project xxx: Exception caught: Get https://some_web.com:8080/v2/alpine/manifests/3.5: unknown: Authentication is required -> [Help 1]`

**Comment by mattnworb** [comment]
should be fixed in v1.0.0
